volume/devicemapper/devicemapper.go

```diff
@@ -935,34 +1031,39 @@ func (v *DeviceMapperVolume) Export(label, parent string, writer io.Writer) erro
 	if err := v.driver.DeviceSet.MountDevice(device, mountpoint, label); err != nil {
 		return err
 	}
-	defer func() {
-		glog.V(2).Infof("Unmounting temporary export device %s", device)
-		// We use the provided UnmountDevice func here, rather than our own
-		// unmount(), because we DO care about Docker's internal bookkeeping
-		// here. Without this, DeviceSet.DeleteDevice will fail.
-		if err := v.driver.DeviceSet.UnmountDevice(device, mountpoint); err != nil {
-			glog.V(2).Infof("Error unmounting (%s): %s", mountpoint, err)
-		}
-		glog.V(2).Infof("Deactivating temporary export device %s", device)
-		v.driver.DeviceSet.Lock()
-		if err := v.driver.deactivateDevice(device); err != nil {
-			glog.V(2).Infof("Error deactivating device (%s): %s", device, err)
-		}
-		v.driver.DeviceSet.Unlock()
-		// Be a good citizen and clean up the mountpoint
-		os.RemoveAll(mountpoint)
-	}()
-
+	defer v.driver.unmountDevice(device, mountpoint)
```

Comment:
The deactivation and stuff is not part of unmounting and probably shouldn't be contained therein. "Unmount" does not always include "deregister the device from the kernel table" and "delete the mount point," it just does in these two cases. Maybe a separate function?